backend-rust/src/graphql_api.rs

```diff
@@ -4319,20 +4583,94 @@ impl BakerStakeIncreased {
 }
 
 #[derive(SimpleObject, serde::Serialize, serde::Deserialize)]
+#[graphql(complex)]
 pub struct ContractInitialized {
-    module_ref:       String,
-    contract_address: ContractAddress,
-    amount:           Amount,
-    init_name:        String,
-    version:          ContractVersion,
-    // TODO: eventsAsHex("Returns the first _n_ elements from the list." first: Int "Returns the
-    // elements in the list that come after the specified cursor." after: String "Returns the last
-    // _n_ elements from the list." last: Int "Returns the elements in the list that come before
-    // the specified cursor." before: String): StringConnection TODO: events("Returns the first
-    // _n_ elements from the list." first: Int "Returns the elements in the list that come after
-    // the specified cursor." after: String "Returns the last _n_ elements from the list." last:
-    // Int "Returns the elements in the list that come before the specified cursor." before:
-    // String): StringConnection
+    module_ref:        String,
+    contract_address:  ContractAddress,
+    amount:            Amount,
+    init_name:         String,
+    version:           ContractVersion,
+    // All logged events by the smart contract during the transaction execution.
+    contract_logs_raw: Vec<Vec<u8>>,
+    // TODO: the send message/input parameter is missing and not exposed by the node and rust SDK
+    // currently, it should be added when available.
+    //
+    // input_parameter:  Vec<u8>,
+}
+
+#[ComplexObject]
+impl ContractInitialized {
+    // TODO: the send message/input parameter is missing and not exposed by the node
+    // and rust SDK currently, it should be added when available.
+    //
+    // async fn message(&self) -> ApiResult<String> {
+    //     // TODO: decode input-parameter/message with schema.
+
+    //     Ok(self.input_parameter)
+    // }
+
+    // TODO: the send message/input parameter is missing and not exposed by the node
+    // and rust SDK currently, it should be added when available.
+    //
+    // async fn message_as_hex(&self) -> ApiResult<String> {
+    // Ok(hex::encode(self.input_parameter)) }
+
+    async fn events_as_hex(&self) -> ApiResult<connection::Connection<String, String>> {
+        let mut connection = connection::Connection::new(true, true);
+
+        self.contract_logs_raw.iter().enumerate().for_each(|(index, log)| {
+            connection.edges.push(connection::Edge::new(index.to_string(), hex::encode(log)));
+        });
+
+        // Nice-to-have: pagination info but not used at front-end currently.
+
+        Ok(connection)
+    }
+
+    async fn events<'a>(
+        &self,
+        ctx: &Context<'a>,
+    ) -> ApiResult<connection::Connection<String, String>> {
+        let pool = get_pool(ctx)?;
+
+        let row = sqlx::query!(
+            r#"
+SELECT
```

Comment:
As in the other places:
```suggestion
        let row = sqlx::query!("
            SELECT
```